source/JsMaterialX/README.md

```diff
@@ -0,0 +1,88 @@
+# Javascript Support
+
+A Javascript package is created from the following modules.
+
+- [JsMaterialXCore](JsMaterialXCore): Contains all of the core classes and util functions.
+- [JsMaterialXFormat](JsMaterialXFormat): Contains the `readFromXmlString` function to read a MaterialX string.
+
+## Generating JS/WASM
+
+### Prerequisites
+
+Make sure to clone the [emsdk repository](https://github.com/emscripten-core/emsdk), install and enable the latest emsdk environment.
+```sh
+# Get the emsdk repo
+git clone https://github.com/emscripten-core/emsdk.git
+
+# Enter that directory
+cd emsdk
+
+# Download and install the latest SDK tools.
+./emsdk install latest
+
+# Make the "latest" SDK "active" for the current user. (writes ~/.emscripten file)
+./emsdk activate latest
+```
+
+For more information follow the steps described in the [emscripten documentation](https://emscripten.org/docs/getting_started/downloads.html). 
+
+### Build
+In the root of directory of this repository run the following:
+
+#### CMake
+The JavasScript library can be built using cmake and make.
+
+1. Activate the emscripten environment
+```sh
+source ../../../../emsdk/emsdk_env.sh
+```
+
+2. Create the `_build` folder
+```sh
+mkdir -p ./_build
+cd ./_build
+```
+
+3. Run cmake and make
+```sh
+# This will generate the release library
+# To build the non minified debug version replace -DMATERIALX_JS_BUILD_TYPE=RELEASE with -DMATERIALX_JS_BUILD_TYPE=DEBUG
+emcmake cmake ../../.. -DMATERIALX_BUILD_JS=ON -DMATERIALX_JS_BUILD_TYPE=RELEASE
+emmake make
+```
+
+#### build.sh
```

Comment:
Used this on Windows. Not sure if calling ninja is appropriate but seems to build. As noted the SDK location as a build option would be good to have.
```
CALL d:/Work/materialx/emsdk/emsdk_env.bat
CALL mkdir _build
CALL cd _build
CALL emcmake cmake ../../.. -DMATERIALX_BUILD_JS=ON -DMATERIALX_JS_BUILD_TYPE=$1
CALL emmake ninja
```

Reply:
I am on Windows. I used the Ubuntu terminal for building. I will see if ninja is good enough for this.

Reply:
I updated the JsMaterialX README on how to build this on windows using ninja.